docs/StardustDocs/topics/readSqlFromCustomDatabase.md

```diff
@@ -0,0 +1,167 @@
+[//]: # (title: How to Extend DataFrame Library for Custom SQL Database Support: Example with HSQLDB)
+
+# How to Extend DataFrame Library for Custom SQL Database Support: Example with HSQLDB
+
+This guide demonstrates how advanced users can extend the Kotlin DataFrame library to support a custom SQL database, using HSQLDB as an example. By following these steps, you will be able to integrate your custom database into the DataFrame library, allowing for seamless DataFrame creation, manipulation, and querying.
+
+---
+
+## Prerequisites
+
+1. **Create a Gradle Project**:  
```

Comment:
Let's mention Gradle plugin "plugins { }" here too, because `@DataSchema` interface requires it to work

Reply:
You mean section
pluginManagement {
    repositories {
        mavenLocal()
        gradlePluginPortal()
    }
} in settings.gradle.kts?

Reply:
But for me schema worked without these plugins

Reply:
```
plugins {
  id("org.jetbrains.kotlinx.dataframe") version "..."
}
```